rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerImplTest.java

```diff
@@ -174,21 +180,81 @@ void logUpdateCollections() {
         assertEvent(eventLogs, 0, BridgeEvents.UPDATE_COLLECTIONS.getEvent(), new Object[]{}, new Object[]{tx.getSender(signatureCache).toString()});
     }
 
-    @Test
-    void logAddSignature() {
-        // Setup logAddSignature params
-        BtcECKey federatorPubKey = BtcECKey.fromPrivate(BigInteger.valueOf(2L));
+    private static Stream<Arguments> logAddSignatureArgProvider() {
+        FederationMember singleKeyFedMember = new FederationMember(
+            BtcECKey.fromPrivate(BigInteger.valueOf(100)),
+            ECKey.fromPrivate(BigInteger.valueOf(100)),
+            ECKey.fromPrivate(BigInteger.valueOf(100))
+        );
+        FederationMember multiKeyFedMember = new FederationMember(
+            BtcECKey.fromPrivate(BigInteger.valueOf(100)),
+            ECKey.fromPrivate(BigInteger.valueOf(200)),
+            ECKey.fromPrivate(BigInteger.valueOf(300))
+        );
+
+        return Stream.of(
+            Arguments.of(singleKeyFedMember, multiKeyFedMember)
+        );
+    }
+
+    @ParameterizedTest()
+    @MethodSource("logAddSignatureArgProvider")
+    void logAddSignature_before_rskip415(FederationMember federationMember) {
+        // Arrange
+        ActivationConfig.ForBlock fingerrootActivations = ActivationConfigsForTest.fingerroot500().forBlock(0);
+        BridgeMainNetConstants bridgeMainNetConstants = BridgeMainNetConstants.getInstance();
+        BlockTxSignatureCache signatureCache = new BlockTxSignatureCache(new ReceivedTxSignatureCache());
+        LinkedList<LogInfo> eventLogs = new LinkedList<>();
+        BridgeEventLogger eventLogger = new BridgeEventLoggerImpl(bridgeMainNetConstants, fingerrootActivations, eventLogs, signatureCache);
+
+
+        BtcECKey federatorBtcPubKey = federationMember.getBtcPublicKey();
+
+        Keccak256 rskTxHash = PegTestUtils.createHash3(1);
+
+        BtcTransaction btcTxMock = mock(BtcTransaction.class);
+        when(btcTxMock.getHash()).thenReturn(BitcoinTestUtils.createHash(1));
+        when(btcTxMock.getHashAsString()).thenReturn(rskTxHash.toHexString());
+
+        // Act
+        eventLogger.logAddSignature(federationMember, btcTxMock, rskTxHash.getBytes());
+
+        // Assert
+        commonAssertLogs(eventLogs);
+        assertTopics(3, eventLogs);
+
+        ECKey federatorECKeyFromBtcPubKey = ECKey.fromPublicOnly(federationMember.getBtcPublicKey().getPubKey());
+        String derivedRskAddress = ByteUtil.toHexString(federatorECKeyFromBtcPubKey.getAddress());
```

Comment:
I think it's best to use hardcoded expected values rather than repeating in the test the same logic as in the function.

My proposal:
Make the test receive `federationMember`, `rskAddressDerivedFromBtcKey`, `rskAddressDerivedFromRskKey`, `activations`

You can then try passing a federation member with the same keys, and one with different keys. And I think that would cover everything. What do you think?

Reply:
Why do I need to pass both rskAddress, from BtcKey and RskKey? I don't understand what should I do with each one passing both, feel kind of lost with that implementation.